File: csharp/2008/fools-gold-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Fool's Gold" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Fool's Gold",
                year: 2008,
                plot: "A new clue to the whereabouts of a lost treasure rekindles a married couple's sense of adventure -- and their estranged romance.",
                rating: 5.7
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Fool's Gold",
                year: 2008
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2008/fools-gold-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Fool's Gold to the database
movies.insert(
    title="Fool's Gold", 
    year=2008, 
    plot="A new clue to the whereabouts of a lost treasure rekindles a married couple's sense of adventure -- and their estranged romance.", 
    rating=5.7
)

# Confirm that the movie was added
movie = movies.select(
    title="Fool's Gold", 
    year=2008
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    